lib/ret_web/channels/hub_channel.ex

```diff
@@ -402,6 +542,66 @@ defmodule RetWeb.HubChannel do
 
   def handle_out("mute", _payload, socket), do: {:noreply, socket}
 
+  defp maybe_store_created_object(socket, payload) do
+    case payload do
+      %{"data" => %{"isFirstSync" => true, "creator" => creator, "template" => template, "networkId" => network_id}} ->
+        created_object = socket.assigns.created_objects |> Enum.find(&(&1.network_id == network_id))
+
+        if created_object == nil do
+          socket
+          |> assign(:created_objects, [
+            %{creator: creator, network_id: network_id, template: template} | socket.assigns.created_objects
+          ])
+        else
+          socket
+        end
+
+      _ ->
+        # This is not a first sync message, so we don't need to do anything.
+        socket
+    end
+  end
+
+  defp should_broadcast_object_manipulation(%{"networkId" => network_id}, type, socket) do
+    account = Guardian.Phoenix.Socket.current_resource(socket)
+    hub = socket |> hub_for_socket
+
+    created_object = socket.assigns.created_objects |> Enum.find(&(&1.network_id == network_id))
+
+    if created_object == nil do
```

Comment:
maybe extract to more overloads (particularly if we introduce a `CreatedObject` struct and call them here to avoid the if/else

```
broadcast_object_manipulation(_socket, _type, nil), do: false
broadcast_object_manipulation(socket, type, %CreatedObject{}) do ...
```